addons/mail/static/src/core/common/message_model.js

```diff
@@ -358,6 +358,53 @@ export class Message extends Record {
         );
     }
 
+    get hasOnlyAttachments() {
+        return this.isBodyEmpty && this.attachment_ids.length > 0;
+    }
+
+    get messagePreviewText() {
+        if (!this.hasOnlyAttachments) {
+            return this.inlineBody || this.subtype_description;
+        }
+        const { attachment_ids: attachments } = this;
+        if (!attachments || attachments.length === 0) {
+            return "";
+        }
+        switch (attachments.length) {
+            case 1:
+                return attachments[0].previewName;
+            case 2:
+                return _t("%(file1)s and %(file2)s", {
+                    file1: attachments[0].previewName,
+                    file2: attachments[1].previewName,
+                    count: attachments.length - 1,
+                });
+            default:
+                return _t("%(file1)s and %(count)s other attachments", {
+                    file1: attachments[0].previewName,
+                    count: attachments.length - 1,
+                });
+        }
+    }
```

Comment:
I would rename to `previewText`, prefix `message` is redundant with the model name.